packages/account/src/providers/provider.ts

```diff
@@ -209,24 +209,19 @@ export type CursorPaginationArgs = {
   before?: string | null;
 };
 
-export type FetchRequestOptions = {
-  method: 'POST';
-  headers: { [key: string]: string };
```

Comment:
I think extending via `requestMiddleware` / `RequestInit` is definitely a better implementation, but was the use case of custom headers not already achievable here?

Reply:
Yes, but the user would have to implement the whole `fetch`, whereas here they can use the existing fetcher and extend the request object as they please:
```ts
// overrides internal fetch
const provider = await Provider.create(url, {fetch: (fetchUrl, request) => {
  request.headers.something = "whaa"
}});

// modify request object but use internal fetch
const provider = await Provider.create(url, {requestMiddleware: (request) => {
  request.headers.something = "whaa"
}});
```

On the topic of custom `fetch`, IMO we should remove the ability for users to provide custom `fetch`es that override ours altogether because we always have to think about that scenario when extending our internal fetching capabilities.